src/eventloop.jl

```diff
@@ -52,7 +57,40 @@ function waitloop()
     end
 end
 
-const idle_delay = Ref(0.5)
-function set_idle_delay(delay_secs::Float64=0.5)
-    idle_delay[] = delay_secs
+# Interact.jl band-aid compatibility code.
+const interact_min_delay = 0.25
+const idle_hold_cond = LevelTrigger()
+const idle_ready_event = LevelTrigger()
+
+"""
+    Increment the semaphore to hold IJulia from sending the "status":"idle" message.
+"""
+function busy()
+    idle_hold_cond += 1
+end
+"""
+    Decrement the semaphore that holds IJulia from sending the "status":"idle" message.
+"""
+function idle()
+    idle_hold_cond -= 1
+end
+"""
+    Wait until IJulia is ready to send an "status":"idle" message.
+"""
+function wait_for_ready()
+    wait(idle_ready_event)
+end
+
+# Define an even more simple level triggered event so that we can easily wait on it.
+import Base.+, Base.-, Base.notify, Base.wait
+mutable struct LevelTrigger
+    state::Ref{Int}
```

Comment:
Oh, yeah, I had totally missed that I had a non-const global in there. It has to be mutable for the struct to be useful, but we could probably just do the integer increment-decrement solution.  I think the complexity-benefit there is better.